infobserve/common/pools.py

```diff
@@ -0,0 +1,58 @@
+""" Singleton Connection Pools for Postgresql and Redis """
+
+import asyncpg
+import aioredis
+from .exceptions import UnitializedRedisConnectionPool
+from . import CONFIG, APP_LOGGER
+
+
+class Singleton(type):
+    _instances = {}  # type: ignore
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
+
+
+class PgPool(metaclass=Singleton):
+
+    async def init_db_pool(self):
+        """Initialize the database connection pool
+        """
+        self.pool = await asyncpg.create_pool(**CONFIG.DB_CONFIG)
+
+    def acquire(self):
+        """ Encapsulate the acquire() method of the asyncpg.Pool object.
+
+        Returns:
+            (asyncpg.connection.Connection): A Connection instance.
+        """
+        return self.pool.acquire()
+
+    async def init_db(self):
+        """Initialize the database.
+        """
+        await self.init_db_pool()
+
+        async with self.acquire() as conn:
+            with open("infobserve-schema.sql") as init_script:
+                try:
+                    async with conn.transaction() as tr:
+                        await conn.execute(init_script.read())
+                        APP_LOGGER.info("Initialized Schema")
+                # The init script should move to schema level and so does this error.
+                except asyncpg.exceptions.DuplicateTableError:
+                    print("Duplicate Table Error Raised the sql init script failed.")
+
+
+class RedisConnectionPool(metaclass=Singleton):
+
+    async def init_redis_pool(self):
+        """Initialize Redis connection pool.
+        """
+        if CONFIG.REDIS_CONFIG:
```

Comment:
Since you're already checking for the existence of `CONFIG.REDIS_CONFIG` in `main`, maybe here you should simply catch a key error, log it and exit the process entirely. This can only happen if the user provided configuration is wrong (I think?)